src/Aspire.Hosting/OpenAI/OpenAIResource.cs

```diff
@@ -0,0 +1,17 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// Represents an OpenAI resource independent of hosting model.
+/// </summary>
+/// <param name="name">The name of the resource.</param>
+public class OpenAIResource(string name) : Resource(name), IResourceWithConnectionString
+{
+    /// <summary>
+    /// Gets the connection string for the OpenAI service.
+    /// </summary>
+    /// <returns>The connection string for the OpenAI service.</returns>
+    string? IResourceWithConnectionString.GetConnectionString() => null;
```

Comment:
> What was the reasoning there?

Our previous discussion that mentioned this property was not used in the code. You found where it was handling the `null` case, and implementing `IResourceWithConnectionString` was required to use `WithReference`.

If you think that it makes sense to have the public property I will add it back and a test case to show how it could be used.

Reply:
Yeah, I think it makes sense to have a public property that can be set by the provisioning code in the future.